apps/macos/metro.config.macos.js

```diff
@@ -7,7 +7,7 @@ const path = require('path');
 const blacklist = require('metro-config/src/defaults/blacklist');
 const {getWatchFolders} = require('@uifabricshared/build-native');
 
-const rnmPath = path.resolve(__dirname, 'node_modules/react-native-macos');
+const rnmPath = path.dirname(require.resolve('react-native-macos/package.json'));
```

Comment:
Actually, why do we need to use dirname at all? Why can't we just do `require.resolve('react-native-macos')`

Reply:
> Was this the main reason `react-native` wasn't properly resolving to `react-native-macos` ?

Yes, `react-native-macos` was actually installed at `../../node_modules/react-native-macos` and not at `./node_modules/react-native-macos`. But we should use `require.resolve` to find the path rather than hard coding it.

> Actually, why do we need to use dirname at all? Why can't we just do `require.resolve('react-native-macos')`

`require.resolve('react-native-macos')` resolves to whatever the `main` field in `package.json` points to. In [this case](https://github.com/microsoft/react-native-macos/blob/45aa5b30b1cfedb07d889da634b164ce75613f90/package.json#L17), it would point to `/path/to/react-native-macos/Libraries/react-native/react-native-implementation.js`. There are no guarantees for what `main` can point to so it's much safer to use `package.json`.